packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/autofocus/AutoFocusFeature.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera.features.autofocus;
+
+import android.hardware.camera2.CameraCharacteristics;
+import android.hardware.camera2.CaptureRequest;
+import io.flutter.plugins.camera.CameraProperties;
+import io.flutter.plugins.camera.features.CameraFeature;
+
+public class AutoFocusFeature extends CameraFeature<FocusMode> {
+  private FocusMode currentSetting = FocusMode.auto;
+
+  // When we switch recording modes we re-create this feature with
+  // the appropriate setting here.
+  private final boolean recordingVideo;
+
+  public AutoFocusFeature(CameraProperties cameraProperties, boolean recordingVideo) {
+    super(cameraProperties);
+    this.recordingVideo = recordingVideo;
+  }
+
+  @Override
+  public String getDebugName() {
+    return "AutoFocusFeature";
+  }
+
+  @Override
+  public FocusMode getValue() {
+    return currentSetting;
+  }
+
+  @Override
+  public void setValue(FocusMode value) {
+    this.currentSetting = value;
+  }
+
+  @Override
+  public boolean checkIsSupported() {
+    int[] modes = cameraProperties.getControlAutoFocusAvailableModes();
+
+    // Check if fixed focal length lens. If LENS_INFO_MINIMUM_FOCUS_DISTANCE=0, then this is fixed.
+    // Can be null on some devices.
+    final Float minFocus = cameraProperties.getLensInfoMinimumFocusDistance();
+    // final Float maxFocus = cameraCharacteristics.get(CameraCharacteristics.LENS_INFO_HYPERFOCAL_DISTANCE);
+
+    // Value can be null on some devices:
+    // https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics#LENS_INFO_MINIMUM_FOCUS_DISTANCE
+    boolean isFixedLength;
+    if (minFocus == null) {
+      isFixedLength = true;
+    } else {
+      isFixedLength = minFocus == 0;
+    }
```

Comment:
Consider a ternary so the assignment and declaration are the same statement: `boolean isFixedLength = minFocus == null ? true : (minFocus == 0);`